app/controllers/api/v1/transactions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Api::V1::TransactionsController < Api::BaseController
+  class GetTransactionError < StandardError; end
+
+  def show
+    channel = get_merchant!(params[:channel])
+    uphold_client = Uphold::Client.new(uphold_connection: channel.publisher.uphold_connection)
+
+    render json: uphold_client.transaction.find(id: params[:id])
+  rescue GetTransactionError => e
+    render json: { errors: e.message }, status: 404
+  rescue Faraday::ResourceNotFound
+    render json: { errors: "Could not find the specified transaction under the publishers account" }, status: 400
+  end
+
+  def get_merchant!(channel)
```

Comment:
Let's rename this `get_channel`
```suggestion
  def get_channel!(channel)
```